src/support/rgw_support_bundle

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+
+# CORTX Python common library.
+# Copyright (c) 2022 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import sys
+import os
+import shutil
+import tarfile
+import errno
+import subprocess
+import argparse
+
+from cortx.utils.errors import BaseError
+from cortx.utils.process import SimpleProcess
+from cortx.utils.conf_store.conf_store import Conf, MappedConf
+
+
+class SupportBundleError(BaseError):
+    """ SupportBundleError exception with error code and output """
+
+    def __init__(self, rc, message, *args):
+        super().__init__(rc, message, *args)
+
+
+class RGWSupportBundle:
+    """Generate rgw specific support bundle."""
+    _conf_file = '/etc/ceph/ceph.conf'
+    _tmp_src = '/tmp/rgw/'
+
+    @staticmethod
+    def generate(bundle_id: str, target_path: str, cluster_conf: str, **filters):
+        """ Generate a tar file. """
+        cortx_config_store = MappedConf(cluster_conf)
+        log_base = cortx_config_store.get('cortx>common>storage>log')
+        machine_id = Conf.machine_id
+
+        if os.path.exists(RGWSupportBundle._tmp_src):
+            RGWSupportBundle._cleanup()
+        os.makedirs(RGWSupportBundle._tmp_src, exist_ok=True)
+        # copy configuration files
+        try:
+            shutil.copyfile(RGWSupportBundle._conf_file,\
+                os.path.join(RGWSupportBundle._tmp_src, 'ceph.conf'))
+        except FileNotFoundError as fe:
+            raise SupportBundleError(errno.ENOENT, "File not found %s", fe)
+
+        # copy rgw client log files
+        log_dir = os.path.join(log_base, f'rgw/{machine_id}')
+        if os.path.exists(log_dir):
```

Comment:
currently rgw_client log file is not in place, so every time it will unnecessarily print on screen 'log file not found'.
@rahul-tripathi-git do we require that?

Reply:
Hi @rahul-tripathi-git , @gargpalak048  
Support bundles are usually invoked by daemon process (earlier it was invoked by csm) 
so our support bundle logs might be ignored in case of production setup.

In case of manual invocation of this support bundle script, this message can be useful hence if we still want to print error message then please mention something like,
 "WARNING: RGW log file does not exists hence skipping log file collection."

Reply:
RGW support bundle will be invoked by Support bundle Framework present in cortx-utils library.
I have added a print statement for this warning msg